pallets/corporate-actions/src/distribution/benchmarking.rs

```diff
@@ -0,0 +1,162 @@
+        PortfolioId::default_portfolio(did),
+        PortfolioId::user_portfolio(did, pnum),
+        vec![MovePortfolioItem { ticker, amount }],
+    )
+    .unwrap();
+}
+
+fn dist<T: Trait>(k: u32) -> (User<T>, CAId, Ticker) {
+    let (owner, ca_id) = setup_ca::<T>(CAKind::UnpredictableBenefit);
+
+    let currency = currency::<T>(&owner);
+    let amount = 1000.into();
+    let pnum = 1.into();
+    portfolio::<T>(&owner, pnum, currency, amount);
+
+    <Module<T>>::distribute(
+        owner.origin().into(),
+        ca_id,
+        Some(pnum),
+        currency,
+        amount,
+        3000,
+        Some(4000),
+    )
+    .unwrap();
+
+    set_ca_targets::<T>(ca_id, k);
+
+    (owner, ca_id, currency)
+}
+
+fn add_investor_uniqueness_claim<T: Trait>(user: &User<T>, scope: Ticker) {
+    use confidential_identity::{compute_cdd_id, compute_scope_id};
+    use frame_system::Origin;
+    use polymesh_primitives::{Claim, InvestorZKProofData, Scope};
+
+    let claim_to = user.did();
+    let investor_uid = user.uid();
+    let proof: InvestorZKProofData = InvestorZKProofData::new(&claim_to, &investor_uid, &scope);
+    let cdd_claim = InvestorZKProofData::make_cdd_claim(&claim_to, &investor_uid);
+    let cdd_id = compute_cdd_id(&cdd_claim).compress().to_bytes().into();
+    let scope_claim = InvestorZKProofData::make_scope_claim(&scope.as_slice(), &investor_uid);
+    let scope_id = compute_scope_id(&scope_claim).compress().to_bytes().into();
+
+    let signed_claim_to = <Origin<T>>::Signed(<Identity<T>>::did_records(claim_to).primary_key);
+
+    <Identity<T>>::add_investor_uniqueness_claim(
+        signed_claim_to.into(),
+        claim_to,
+        Claim::InvestorUniqueness(Scope::Ticker(scope), scope_id, cdd_id),
+        proof,
+        None,
+    )
+    .unwrap();
+}
+
+fn prepare_transfer<T: Trait + pallet_compliance_manager::Trait>(
+    k: u32,
+) -> (User<T>, User<T>, CAId) {
+    let (owner, ca_id, currency) = dist::<T>(k);
+
+    <pallet_timestamp::Now<T>>::set(3000.into());
+
+    let holder = user::<T>("holder", SEED);
+    add_investor_uniqueness_claim(&owner, currency);
+    add_investor_uniqueness_claim(&holder, currency);
+    <ComplianceManager<T>>::add_compliance_requirement(
+        owner.origin().into(),
+        currency,
+        vec![],
+        vec![],
+    )
+    .unwrap();
+
+    (owner, holder, ca_id)
+}
+
+benchmarks! {
+    where_clause { where T: pallet_compliance_manager::Trait }
+
+    _ {}
+
+    distribute {
+        let (owner, ca_id) = setup_ca::<T>(CAKind::UnpredictableBenefit);
+        let currency = currency::<T>(&owner);
+        let amount = 1000.into();
+        let pnum = 1.into();
+        portfolio::<T>(&owner, pnum, currency, amount);
+    }: _(owner.origin(), ca_id, Some(pnum), currency, amount, 3000, Some(4000))
+    verify {
+        ensure!(<Distributions<T>>::get(ca_id).is_some(), "distribution not created");
+    }
+
+    // TODO(Centril): make this work with WASM execution.
```

Comment:
Is it done?

Reply:
No, not intended to be fixed right now. We'd have to hardcode stuff unfortunately. I'd like to get the main structure merged first.